query/query.go

```diff
@@ -844,6 +914,115 @@ func (q *ParsedGroupedQuery) computeNewCalculationResultArrayFn() calculation.Ne
 	}
 }
 
```

Comment:
meta, this file is 1000+ lines, could break it up into 4 separate files? one for parsing, and one of each for raw, group, and time bucket?

Reply:
Sounds good.